ui/src/router/details/RouterDetailsView.js

```diff
@@ -42,12 +45,19 @@ export const RouterDetailsView = ({ projectId, routerId, ...props }) => {
   }, [fetchRouterDetails, props.location.state]);
 
   useEffect(() => {
-    if (router.status === Status.PENDING.toString()) {
+    if (router.status?.toString() === Status.PENDING.toString()) {
       const interval = setInterval(fetchRouterDetails, 5000);
       return () => clearInterval(interval);
     }
   }, [router.status, fetchRouterDetails]);
 
+  useEffect(() => {
+    // Parse router details
```

Comment:
Every time the router details change, parse the data for use and store in the local state.